### PR TITLE
feat: added missing requestPermissions and getPermissions API methods

### PR DESCRIPTION
**Pull request by shanejonas**
**Description**
Added missing methods `wallet_requestPermissions` and `wallet_getPermissions` to the mobile API.

This brings it in line with the extension API so that dapps that build against requestPermissions don't have to have extra logic to do `requestAccounts` instead.

fixes https://github.com/MetaMask/MetaMask-planning/issues/861

**Screenshots/Recordings**

some screenshots of it working in the new api playground:

### `wallet_requestPermissions`
image
image
#### error case:
image

### `wallet_getPermissions`

image

### CODE REVIEW

**Comment by sethkfman** [comment]
@shanejonas I would update this PR as a feature these are new methods implemented.

File: app/core/RPCMethods/RPCMethodMiddleware.ts [hunk 1]
```diff
@@ -223,7 +227,61 @@ export const getRpcMethodMiddleware = ({
...
+        new Promise<any>((resolve) => {
+          getPermissionsHandler.implementation(
+            req,
+            res,
+            next,
+            () => {
+              resolve(undefined);
+            },
+            {
+              requestPermissionsForOrigin:
```
Comment (Cal-L) [inline app/core/RPCMethods/RPCMethodMiddleware.ts:243]: Doesn't look like this is used in the `getPermissionsHandler` logic in the controller

File: app/core/RPCMethods/RPCMethodMiddleware.ts [hunk 2]
```diff
@@ -223,7 +227,61 @@ export const getRpcMethodMiddleware = ({
...
+                }
+                resolve(undefined);
+              },
+              {
+                requestPermissionsForOrigin:
+                  Engine.context.PermissionController.requestPermissions.bind(
+                    Engine.context.PermissionController,
+                    { origin: hostname },
+                  ),
+                getPermissionsForOrigin:
```
Comment (Cal-L) [inline app/core/RPCMethods/RPCMethodMiddleware.ts:275]: Doesn't look like this is used in the `requestPermissionsHandler` logic in the controller

File: app/core/RPCMethods/RPCMethodMiddleware.ts (hunk 1 above)
Comment (Cal-L) [reply app/core/RPCMethods/RPCMethodMiddleware.ts:243]: But the input type expects it - Looks like the input type on the controller level needs to be updated?

**Review by Cal-L — COMMENTED** [review]
Left some comments

**Review by Cal-L — DISMISSED** [review]
LGTM, just need to fix a lint issue

**Pull request closed without merging**